src/uwtools/drivers/cdeps.py

```diff
@@ -80,57 +45,65 @@ def atm_nml(self):
+        fn = "datm.streams"
+        yield self._taskname(f"stream file {fn}")
         path = self._rundir / fn
         yield asset(path, path.is_file)
-        yield None
-        path.parent.mkdir(parents=True, exist_ok=True)
-        self._model_namelist_file("ocn_in", path)
+        template_file = self._driver_config["atm_streams"]["template_file"]
+        yield file(path=Path(template_file))
+        self._model_stream_file("atm_streams", path, template_file)
+
+    @tasks
+    def ocn(self):
+        """
+        Create data ocean configuration with all required content.
+        """
+        yield self._taskname("data atmosphere configuration")
+        yield [
+            self.ocn_nml(),
+            self.ocn_stream(),
+        ]
 
     @task
-    def atm_stream(self):
+    def ocn_nml(self):
         """
-        It creates data atmosphere stream config file (datm.streams).
+        Create data ocean Fortran namelist file (docn_in).
         """
-        fn = "datm.streams"
+        fn = "docn_in"
         yield self._taskname(f"namelist file {fn}")
         path = self._rundir / fn
         yield asset(path, path.is_file)
-        temp_path = self._driver_config["atm_streams"]["base_file"]
-        yield file(path=Path(temp_path))
-        self._model_stream_file("atm_streams", path, temp_path)
+        yield None
+        path.parent.mkdir(parents=True, exist_ok=True)
+        self._model_namelist_file("ocn_in", path)
 
     @task
     def ocn_stream(self):
         """
-        It creates data ocean stream config file (docn.streams).
+        Create data ocean stream config file (docn.streams).
         """
         fn = "docn.streams"
-        yield self._taskname(f"namelist file {fn}")
+        yield self._taskname(f"stream file {fn}")
         path = self._rundir / fn
         yield asset(path, path.is_file)
-        temp_path = self._driver_config["ocn_streams"]["base_file"]
-        yield file(path=Path(temp_path))
-        self._model_stream_file("ocn_streams", path, temp_path)
+        template_file = self._driver_config["ocn_streams"]["template_file"]
+        yield file(path=Path(template_file))
+        self._model_stream_file("ocn_streams", path, template_file)
 
-    def _model_namelist_file(self, group: str, path: str):
-        self._create_user_updated_config(
-            config_class=NMLConfig,
-            config_values=self._driver_config[group],
-            path=path,
-        )
-
-    def _model_stream_file(self, group: str, path: str, template: str):
-        render(
-            input_file=Path(template),
-            output_file=path,
-            values_src=self._driver_config[group],
-        )
+    @tasks
+    def provisioned_rundir(self):
+        """
+        Run directory provisioned with all required content.
+        """
+        yield self._taskname("provisioned run directory")
+        yield [
+            self.atm(),
+            self.ocn(),
+        ]
```

Comment:
This method is required by the `Assets` base class that this driver ultimately inherits from. It might not be useful for CDEPS. If it's too confusing, we could reconsider making this method mandatory,  Please let me know if you think that this method should _not_ exist and I can take care of it later after the main PR merges to `uwtools` `main`.